utils/utils.go

```diff
@@ -0,0 +1,19 @@
+package utils
+
+import (
+	"encoding/json"
+	"log"
+	"net/http"
+)
+
+// Adapted from https://github.com/prometheus-community/prom-label-proxy/blob/02d43edb82b7d139f4a4e41912ad903bff46d5c4/injectproxy/utils.go#L22
+func PrometheusAPIError(w http.ResponseWriter, errorMessage string, code int) {
+	w.Header().Set("Content-Type", "application/json; charset=utf-8")
+	w.WriteHeader(code)
+
+	res := map[string]string{"status": "error", "errorType": "observatorium-api", "error": errorMessage}
+
+	if err := json.NewEncoder(w).Encode(res); err != nil {
```

Comment:
Optional: My personal preference only - encode into JSON and then use `http.Error` to set code, headers etc.; that way we can 'outsource' it to the `http` package and not do it ourselves by hand

Reply:
Discussed with @matej-g, so using http.Error isn't suitable here as it always sends a [plaintext response](https://cs.opensource.google/go/go/+/refs/tags/go1.18.3:src/net/http/server.go;l=2093). So resolving this! 🙂